src/Raven.Server/Documents/Sharding/ShardingRecordExtensions.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Collections.Generic;
+using Raven.Client.ServerWide;
+using Raven.Client.ServerWide.Sharding;
+using Sparrow.Utils;
+using Voron;
+
+namespace Raven.Server.Documents.Sharding;
+
+public static class ShardingRecordExtensions
+{
+    public static List<ShardBucketRange> For(this ShardingRecord record, string id)
+    {
+        if (record.Prefixed == null)
+            return record.ShardBucketRanges;
+
+        foreach (var kvp in record.Prefixed)
+        {
+            if (id.StartsWith(kvp.Key, StringComparison.OrdinalIgnoreCase))
+                return kvp.Value;
+        }
+
+        return record.ShardBucketRanges;
```

Comment:
Just to mention that if we want to shard a specific collection.
e.g. `Orders` I will set the prefix `orders/` and the `ShardBucketRanges` will contain a single entry and map to a specific shard, right?

Reply:
Yes